backend/src/services/trading_service.py

```
@@ -175,57 +195,141 @@ async def execute_sell_order(self, token_address: str, amount_tokens: float, sli
+                            result = await self.execute_sell_order(order['token_address'], balance)
+                        else:
+                            result = {"success": False, "message": "No tokens to sell"}
+
+                    if result.get("success"):
+                        update_limit_order_status(order['id'], 'executed')
+                        if self.socketio:
+                            self.socketio.emit('limit_order_executed', {**order, "execution_price": current_price})
+                    else:
+                        logger.error(f"Failed to execute limit order {order['id']}: {result.get('message')}")
+
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)